bundle-workflow/src/test_workflow/test_recorder.py

```diff
@@ -3,34 +3,133 @@
 # The OpenSearch Contributors require contributions made to
 # this file be licensed under the Apache-2.0 license or a
 # compatible open source license.
-
 import logging
+import os
+import shutil
+import tempfile
+
+import yaml
 
 
 class TestRecorder:
-    def __init__(self, location):
-        self.location = location
-        logging.info(f"TestRecorder storing results in {location}")
+    ACCEPTED_TEST_TYPES = ["integ-test", "bwc-test", "perf-test"]
+
+    def __init__(self, test_run_id, test_type, location=None):
+        self.test_type = test_type
+        self.test_run_id = test_run_id
+
+        if self.test_type not in self.ACCEPTED_TEST_TYPES:
+            raise ValueError(f"test_type is invalid. Acceptable test_type are {self.ACCEPTED_TEST_TYPES}")
+
+        if location is None:
```

Comment:
We have to clean up the directory when TestRecorder goes out of scope. Tmp directory gives us a great feature which takes care of it. It's better to clean it up by ourself, we do not want to leave them on the system. 

Lets only use temporary directory as the only option for TestRecorder and when it goes out of scope let it destroy the directory and add a keep parameter if caller wants the directory to be preserved.
Anyway we don't have a use case yet for a custom directory, avoids the hassle of cleaning up.

Reply:
Makes sense. Regarding clean up. We cannot clean up the directory unless publisher is done publishing the result. So would it make sense that publisher deletes the directory when it is done publishing? or is there any other way?

Reply:
The pattern I use is to have publisher create a temporary directory, tell the published to put files there via parameters, and never delete anything explicitly. Temporary directories are cleaned up automatically.

Reply:
lets leave the ownership to the test suite runners. They use test recorder, TestRecorder records them and deletes them if it goes out of scope. 
Any way TestPublisher(aka TestResults) consumes TestRecorder so it is implicitly taken care of.

Reply:
Sarat suggests that the topmost caller creates a temporary dir and it's passed down to anyone that needs to write files.

Reply:
Yeah that is happening right now. Will see how to clean_up when the test_recorder object goes out of scope